www/js/splash/storedevicesettings.js

```diff
@@ -44,16 +45,17 @@ angular.module('emission.splash.storedevicesettings', ['emission.plugin.logger',
       Logger.log("storedevicesettings startup done");
     });
 
-    $rootScope.$on(StartPrefs.CONSENTED_EVENT, function(event, data) {
-      console.log("got consented event "+JSON.stringify(event.name)
-                      +" with data "+ JSON.stringify(data));
-      if (StartPrefs.isIntroDone()) {
+    storedevicesettings.afterConsent = function() {
+      console.log("in storedevicesettings, executing after consent is received");
+      readIntroDone().then((intro_done) => {
+        if (intro_done) {
           console.log("intro is done -> reconsent situation, we already have a token -> register");
           storedevicesettings.storeDeviceSettings();
       }
-    });
+      })
+    }
 
-    $rootScope.$on(StartPrefs.INTRO_DONE_EVENT, function(event, data) {
+    $rootScope.$on(startPrefs.INTRO_DONE_EVENT, function(event, data) {
           console.log("intro is done -> original consent situation, we should have a token by now -> register");
        storedevicesettings.storeDeviceSettings();
     });
```

Comment:
It looks like `INTRO_DONE_EVENT` is no longer emitted from anywhere. So I don't think you need to listen for it anymore, but I *do* think you need to find a suitable replacement.

From analyzing these 2 events (CONSENTED_EVENT and INTRO_DONE_EVENT) and seeing how the old StartPrefs worked, here's the high-level takeaway I'm getting.
When either:
- a) the intro gets done
- b) the user reconsents (consent gets done and intro was already done)

We call  `storedevicesettings.storeDeviceSettings()` and `pushnotify.registerPush`.

So I think the `afterConsent` mechanism you made should be something more like `afterIntroOrReconsent`, and I think SaveQrPage is the wrong place to do it.
I'm not sure if reconsent is still a thing or will be a thing in the future (I saw there was recent discussion on this). If not, I suppose you'll only need to consider (a) and not (b)

Reply:
We're taking out reconsent for now, as decided in [this comment](https://github.com/e-mission/e-mission-docs/issues/1016#issuecomment-1778476728). Combining the mechanism makes sense, I'll work on combining things and finding a better place for it.

Reply:
>a) the intro gets done

One place to catch this would be to add the functionality to the `markIntroDone` method in onboarding helper 

>b) the user reconsents (consent gets done and intro was already done)

I could add the after consent part to `markConsented` in `startPrefs`  - (if Intro is done) this would be a very similar place functionally, but maybe better organizationally? 

@JGreenlee do those places sound better?

Reply:
Yes, those 2 places seem appropriate to me

Reply:
Done! I tested by logging in and out of the emulator and seeing the log statement for the "after intro events" and not the log statement for the "reconsent events". This is expected behavior since reconsent is not what's happening on login, and we don't currently have a reconsent scenario. I'll add some git comments to the new code to flag the changes.